pkg/crc/api/handlers.go

```diff
@@ -13,13 +14,17 @@ import (
 	"github.com/code-ready/crc/pkg/crc/version"
 )
 
+var mutex sync.Mutex
+
 func statusHandler(_ ArgsType) string {
 	statusConfig := machine.ClusterStatusConfig{Name: constants.DefaultName}
 	clusterStatus, _ := machine.Status(statusConfig)
 	return encodeStructToJson(clusterStatus)
 }
 
 func stopHandler(_ ArgsType) string {
+	mutex.Lock()
+	defer mutex.Unlock()
 	stopConfig := machine.StopConfig{
```

Comment:
at this point, I think I'd prefer that we achieve the serialization at the api.go level rather than adding mutexes to the `machine` package.